src/algebra/hom/centroid.lean

```diff
@@ -0,0 +1,299 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Christopher Hoskin
+-/
+import algebra.group.pi
+import algebra.group_power.lemmas
+import algebra.hom.group_instances
+
+/-!
+# Centroid homomorphisms
+
+Let `A` be a (non unital, non associative) algebra. The centroid of `A` is the set of linear maps
+`T` on `A` such that `T` commutes with left and right multiplication, that is to say, for all `a`
+and `b` in `A`,
+$$
+T(ab) = (Ta)b, T(ab) = a(Tb).
+$$
+In mathlib we call elements of the centroid "centroid homomorphisms" (`centroid_hom`) in keeping
+with `add_monoid_hom` etc.
+
+We use the `fun_like` design, so each type of morphisms has a companion typeclass which is meant to
+be satisfied by itself and all stricter types.
+
+## Types of morphisms
+
+* `centroid_hom`: Maps which preserve left and right multiplication.
+
+## Typeclasses
+
+* `centroid_hom_class`
+
+## References
+
+* [Jacobson, Structure of Rings][Jacobson1956]
+* [McCrimmon, A taste of Jordan algebras][mccrimmon2004]
+
+## Tags
+
+centroid
+-/
+
+open function
+
+variables {F α : Type*}
+
+section
```

Comment:
You don't need this `section` because you won't declare any other `structure` in this file.

Reply:
Removed.